beamer/agent/chain.py

```diff
@@ -328,6 +328,7 @@ def process_claims(context: Context) -> None:
             if _l1_resolution_threshold_reached(claim, context):
                 maybe_resolve(claim, context)
             else:
+                maybe_prove(claim, context)
```

Comment:
I think `maybe_prove` must be called unconditionally. It could be the case that a threshold is reached but prove was not called yet.

In this context I would pull the condition to be a Optimism chain out of the maybe_prove and put it here. This way it gets clear to the code reader that it is an Optimism depended step.

Reply:
IMO we should pull the `maybe_prove` out of the condition above and call it unconditionally. The check if the message is proved is already in maybe_prove

Reply:
> IMO we should pull the `maybe_prove` out of the condition above and call it unconditionally. The check if the message is proved is already in maybe_prove

I agree with this. :+1:
We could also think of this separate "proving" step as a fine-grained step where OP does something and for everything else it's a no-op. (Could also happen that future supported chains need this...)